packages/vx-axis/src/axis/Axis.tsx

```diff
@@ -1,216 +1,86 @@
-import getLabelTransform from '../utils/labelTransform';
-import ORIENT from '../constants/orientation';
-import toString from '../utils/toString';
-import toNumberOrUndefined from '../utils/toNumberOrUndefined';
-import { SharedAxisProps, AxisOrientation } from '../types';
+import { getTicks, coerceNumber } from '@vx/scale';
+import { SharedAxisProps, AxisScale } from '../types';
+import AxisRenderer from './AxisRenderer';
+import getTickPosition from '../utils/getTickPosition';
+import getTickFormatter from '../utils/getTickFormatter';
+import createPoint from '../utils/createPoint';
+import Orientation from '../constants/orientation';
 
-export type AxisProps<ScaleInput> = SharedAxisProps<ScaleInput> & {
-  orientation?: AxisOrientation;
+export type AxisProps<Scale extends AxisScale> = SharedAxisProps<Scale> & {
+  orientation?: Orientation;
 };
 
-export default function Axis<ScaleInput>({
-  children,
+export default function Axis<Scale extends AxisScale>({
+  children = AxisRenderer,
   axisClassName,
-  axisLineClassName,
   hideAxisLine = false,
   hideTicks = false,
   hideZero = false,
-  label = '',
-  labelClassName,
-  labelOffset = 14,
-  labelProps = {
-    textAnchor: 'middle',
-    fontFamily: 'Arial',
-    fontSize: 10,
-    fill: '#222',
-  },
   left = 0,
   numTicks = 10,
-  orientation = ORIENT.bottom,
+  orientation = Orientation.bottom,
   rangePadding = 0,
   scale,
-  stroke = '#222',
-  strokeWidth = 1,
-  strokeDasharray,
-  tickClassName,
   tickFormat,
-  tickLabelProps = (/** tickValue, index */) => ({
-    textAnchor: 'middle',
-    fontFamily: 'Arial',
-    fontSize: 10,
-    fill: '#222',
-  }),
   tickLength = 8,
-  tickStroke = '#222',
-  tickTransform,
   tickValues,
-  tickComponent,
   top = 0,
-}: AxisProps<ScaleInput>) {
-  const values =
-    tickValues ||
-    (scale.ticks
-      ? scale.ticks(numTicks)
-      : scale
-          .domain()
-          .filter(
-            (_, index, arr) =>
-              numTicks == null ||
-              arr.length <= numTicks ||
-              index % Math.round((arr.length - 1) / numTicks) === 0,
-          ));
-  const format = tickFormat || (scale.tickFormat ? scale.tickFormat() : toString);
+  ...restProps
+}: AxisProps<Scale>) {
+  const format = tickFormat ?? getTickFormatter(scale);
 
-  const range = scale.range();
-  const range0 = Number(range[0]) + 0.5 - rangePadding;
-  const range1 = Number(range[range.length - 1]) + 0.5 + rangePadding;
+  const isLeft = orientation === Orientation.left;
+  const isTop = orientation === Orientation.top;
+  const horizontal = isTop || orientation === Orientation.bottom;
 
-  const isLeft = orientation === ORIENT.left;
-  const isTop = orientation === ORIENT.top;
-  const axisIsHorizontal = isTop || orientation === ORIENT.bottom;
+  const tickPosition = getTickPosition(scale);
   const tickSign = isLeft || isTop ? -1 : 1;
 
-  const position = center(scale.copy());
-
-  const axisFromPoint = new Point({
```

Comment:
I extracted the logic for flipping `x` and `y` into util function `createPoint`.